aceto.py

```diff
@@ -73,16 +74,29 @@ def __init__(self, args):
             except AttributeError:
                 pass
 
-    def load_code(self, filename):
+    def load_code(self, filename, linear_mode=False):
         self.code = []
         with open(filename, encoding=self.encoding) as f:
-            for line in reversed(f.readlines()):
-                self.code.append(list(line.rstrip("\n")))
-        self.p = ceil(
-                log2(
-                    max([len(self.code), max(len(line) for line in self.code)])
-                    )
-                )
+            if linear_mode:
+                code_helper = defaultdict(lambda:defaultdict(str))
+                chars = ''.join(char for line in f for char in line.strip())
+                self.p = ceil(log2(ceil(len(chars)**.5)))
+                for step, char in enumerate(chars):
+                    y, x = hilbert.coordinates_from_distance(
+                            step,
+                            self.p,
+                            N=2,
+                            )
+                    code_helper[x][y] = char
+                self.code = code_helper
```

Comment:
I realize this isn't the cleanest of ways: `self.code` is now either a list of lists or a dict of dicts, but the underlying protocol shouldn't care. In fact, it might be a good idea to turn the list-of-lists `self.code` object into the dict of dicts objects even in normal mode, because it somehow feels more correct.